x/posts/internal/keeper/keeper_reactions.go

```diff
@@ -0,0 +1,150 @@
+	if !reactions.ContainsReactionFrom(user, value) {
+		return fmt.Errorf("cannot remove the reaction with value %s from user %s as it does not exist",
+			value, user)
+	}
+
+	// Remove and save the reactions list
+	if newLikes, edited := reactions.RemoveReaction(user, value); edited {
+		if len(newLikes) == 0 {
+			store.Delete(key)
+		} else {
+			store.Set(key, k.Cdc.MustMarshalBinaryBare(&newLikes))
+		}
+	}
+
+	return nil
+}
+
+// GetPostReactions returns the list of reactions that has been associated to the post having the given id
+// nolint: interfacer
+func (k Keeper) GetPostReactions(ctx sdk.Context, postID types.PostID) types.PostReactions {
+	store := ctx.KVStore(k.StoreKey)
+
+	var reactions types.PostReactions
+	k.Cdc.MustUnmarshalBinaryBare(store.Get(types.PostReactionsStoreKey(postID)), &reactions)
+	return reactions
+}
+
+// GetReactions allows to returns the list of reactions that have been stored inside the given context
+func (k Keeper) GetReactions(ctx sdk.Context) map[types.PostID]types.PostReactions {
+	store := ctx.KVStore(k.StoreKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.PostReactionsStorePrefix)
+
+	reactionsData := map[types.PostID]types.PostReactions{}
+	for ; iterator.Valid(); iterator.Next() {
+		var postLikes types.PostReactions
+		k.Cdc.MustUnmarshalBinaryBare(iterator.Value(), &postLikes)
+		idBytes := bytes.TrimPrefix(iterator.Key(), types.PostReactionsStorePrefix)
+		postID, err := types.ParsePostID(string(idBytes))
+		if err != nil {
+			// This should never verify
+			panic(err)
+		}
+
+		reactionsData[postID] = postLikes
+	}
+
+	return reactionsData
+}
+
+// -------------
+// --- Reactions
+// -------------
+
+// RegisterReaction allows to register a new reaction for later reference
+func (k Keeper) RegisterReaction(ctx sdk.Context, reaction types.Reaction) {
+	store := ctx.KVStore(k.StoreKey)
+	key := types.ReactionsStoreKey(reaction.ShortCode, reaction.Subspace)
+	store.Set(key, k.Cdc.MustMarshalBinaryBare(&reaction))
+}
+
+// DoesReactionForShortCodeExist checks whether a reaction already exists for the given shortCode, returning it if it does.
+func (k Keeper) DoesReactionForShortCodeExist(ctx sdk.Context, shortcode string, subspace string) (reaction *types.Reaction, exist bool) {
+	store := ctx.KVStore(k.StoreKey)
+	key := types.ReactionsStoreKey(shortcode, subspace)
+
+	bz := store.Get(key)
+
+	if bz != nil {
+		k.Cdc.MustUnmarshalBinaryBare(bz, &reaction)
+		return reaction, true
+	}
+
+	return nil, false
+}
+
+// ListReactions returns all the registered reactions
+func (k Keeper) ListReactions(ctx sdk.Context) types.Reactions {
+	store := ctx.KVStore(k.StoreKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.ReactionsStorePrefix)
+
+	reactions := make(types.Reactions, 0)
+
+	for ; iterator.Valid(); iterator.Next() {
+		var reaction types.Reaction
+		k.Cdc.MustUnmarshalBinaryBare(iterator.Value(), &reaction)
+		reactions = append(reactions, reaction)
+	}
+
+	return reactions
```

Comment:
We can remove the middle `reactions :=` initialization by declaring it as the return value. Also, make sure to close the `Iterator`.

```suggestion
func (k Keeper) ListReactions(ctx sdk.Context) (reactions types.Reactions) {
	store := ctx.KVStore(k.StoreKey)
	iterator := sdk.KVStorePrefixIterator(store, types.ReactionsStorePrefix)
	defer iterator.Close()
	
	for ; iterator.Valid(); iterator.Next() {
		var reaction types.Reaction
		k.Cdc.MustUnmarshalBinaryBare(iterator.Value(), &reaction)
		reactions = append(reactions, reaction)
	}

	return reactions
```